[BUG] Compilation error: Float64 pattern not covered in safetensors.rs

## Bug Description

mlx-rs fails to compile when using the `safetensors` feature due to non-exhaustive pattern matching for `Dtype::Float64`.

## Environment
- **mlx-rs version:** 0.25.2 (both crates.io and git main @ 7c77f32)
- **Rust version:** rustc 1.91.1 (ed61e7d7e 2025-11-07) (Homebrew)
- **Platform:** macOS 26.1, Apple Silicon (arm64)
- **Features:** `safetensors`

## Error Message

```
error[E0004]: non-exhaustive patterns: `dtype::Dtype::Float64` not covered
  --> mlx-rs/src/array/safetensors.rs:42:19
   |
42 |             match value.dtype() {
   |                   ^^^^^^^^^^^^^ pattern `dtype::Dtype::Float64` not covered
   |
note: `dtype::Dtype` defined here
  --> mlx-rs/src/dtype.rs:21:14
   |
21 |     pub enum Dtype {
   |              ^^^^^
...
56 |         Float64 = mlx_sys::mlx_dtype__MLX_FLOAT64,
   |         ------- not covered
```

## Reproduction

Add mlx-rs as a dependency with safetensors feature:

```toml
[dependencies]
mlx-rs = { version = "0.25", features = ["safetensors"] }
```

Then run:
```bash
cargo check
```

## Expected Behavior

Code should compile successfully.

## Actual Behavior

Compilation fails with non-exhaustive pattern match error.

## Suggested Fix

Add a match arm for `Dtype::Float64` in `mlx-rs/src/array/safetensors.rs:42`:

```rust
match value.dtype() {
    // ... existing patterns ...
    Dtype::Float64 => {
        // Handle Float64 conversion
    }
    // ... rest of patterns ...
}
```

## Impact

This blocks any project from using mlx-rs with the safetensors feature, which is critical for loading HuggingFace models.

## Workaround

None currently available without patching mlx-rs locally.

--- a/mlx-rs/src/array/safetensors.rs
+++ b/mlx-rs/src/array/safetensors.rs
@@ -85,6 +85,10 @@ impl<'a> TryFrom<&'a Array> for TensorView<'a> {
                     let data = value.as_slice::<f32>();
                     cast_slice(data)
                 },
+                Dtype::Float64 => {
+                    let data = value.as_slice::<f64>();
+                    cast_slice(data)
+                },
                 Dtype::Bfloat16 => {
                     let data = value.as_slice::<half::bf16>();
                     let bits: &[u16] = transmute(data);

